app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -89,88 +90,117 @@ class DrawActivity : AppCompatActivity() {
-                            viewModel.addLayer(layerName)
-                        } else viewModel.dialogCompleted()
+                            viewModel.addToList(layerName)
+                        } else {
+                            viewModel.dialogCompleted()
+                        }
                     },
                 )
-            } else if (!it.isDialogOpen) {
-                isAddDialogOpen = false
-            }
-            if (it.isLayerSheetOpen && !isLayerListViewSheetOpen) {
+            } else isAddDialogOpen = false
+            if (it.isLayerListViewSheetOpen && !isLayerListViewSheetOpen) {
                 isLayerListViewSheetOpen = true
-                viewModel.setLayerListViewState()
-                bottomSheetLambda = dialogUtils.layerListBottomSheet(context = this,
+                onLayerListUpdate = dialogUtils.layerListBottomSheet(context = this,
                     layerList = it.layerList,
-                    onDialogComplete = { viewModel.dialogCompleted() },
+                    onSheetComplete = {
+                        isLayerListViewSheetOpen = false
+                        viewModel.dialogCompleted()
+                    },
                     onDelete = { index ->
-                        viewModel.deleteLayer(index)
-                        if (!viewModel.isLayerListEmpty()) {
-                            bottomSheetLambda?.invoke(index, null, DialogOperation.DELETE)
+                        viewModel.setCurrentIndex(index)
+                        viewModel.deleteDialogOpened()
+                        isDeleteDialogOpen = true
+                        if (it.isDeleteDialogOpen) {
+                            dialogUtils.confirmationDialog(context = this@DrawActivity,
+                                title = this@DrawActivity.getString(R.string.deleteLayerDialogTitle),
+                                message = this@DrawActivity.getString(
+                                    R.string.deleteLayerDialogBody, it.layerList[index]
+                                ),
+                                onConfirm = {
+                                    viewModel.setCurrentIndex(index)
+                                    viewModel.removeFromList(index)
+                                    onLayerListUpdate?.invoke(
+                                        viewModel.getCurrentIndex(), null, DialogOperation.DELETE
+                                    )
+                                    viewModel.dialogCompleted()
+                                },
+                                onCancel = {
+                                    viewModel.dialogCompleted()
+                                })
                         }
                     },
                     onEdit = { index ->
                         isEditDialogOpen = true
-                        viewModel.setDialogState()
-                        viewModel.setEditIndex(index)
+                        viewModel.editDialogOpened()
```

Comment:
The view should be calling something like `onEditClicked`, it shouldn't know there's a dialog involved

Reply:
adjusted the name of this